lib/ClangImporter/ImportDecl.cpp

```diff
@@ -8437,6 +8438,107 @@ ClangImporter::Implementation::importSwiftAttrAttributes(Decl *MappedDecl) {
   }
 }
 
+namespace {
+class PointerParamInfo {
+public:
+  virtual void print(clang::ASTContext &ctx, llvm::raw_ostream &out) const = 0;
+  virtual ~PointerParamInfo() {}
+};
+
+class CountedByParam : public PointerParamInfo {
```

Comment:
I presume `Clang` also has a representation of this information. Any reason we cannot reuse Clang's classes for that instead of rolling our own? E.g., we could just reuse `CountAttributedType` and add a custom print function for it.

Reply:
We could do that, but we would need to make it a pair with both `CountAttributedType` and pointer index. When we add `DynamicRangeType` we also need the pointer index for the end pointer. Another option would be to not build a `SmallVector` at all, and print on the first pass. Not sure whether initialising `MacroString` and the outputstream when there are no bounds annotations to process is preferable. I guess if it turns out to be a problem we could mark functions for later bounds importing when we initially process parameters and encounter a bounds attribute?

Reply:
Added a commit with another approach, please let me know what you think